src/content/blog/rust-based-javascript-linters-lack-typed-linting-right-now/index.mdx

```diff
@@ -0,0 +1,151 @@
+But even if that option works perfectly then type-checked linting will by design be orders of magnitude slower than traditional linting.
+
+Our experience has been that the benefits of type-checked linting almost always far outweigh the drawbacks of slower performance.
+Most of the time, when we've seen projects with slow type-checked linting, the root cause was either a misconfiguration of typescript-eslint or [slow TypeScript types](https://github.com/microsoft/TypeScript/wiki/Performance).
+
+## Rust-Based Linters and Type Checking
+
+**No Rust-based linter has integrated with TypeScript's type checking APIs yet**.
+That means no Rust-based linter is a full replacement for ESLint + typescript-eslint.
+
+I'm not saying you shouldn't use a Rust-based linter: if you don't want any of the type-checked lint rules, then sure, switching over is great.
+But I strongly recommend you look through at least the [recommended type-checked rules in typescript-eslint](https://typescript-eslint.io/rules/?=recommended-typeInformation) to understand what you're missing first.
+
+### Integrating Type-Checked Linting and Rust-Based Linters
+
+Right now, TypeScript is the only tool that can reliably provide type-checked linting for TypeScript code.
+TypeScript is written in TypeScript, so its type checking runs at JavaScript speed.
+
+In order to work with type checking, a Rust linter would have to either:
+
+- Take the performance hit of calling to TypeScript's JavaScript-speed type-checking APIs
+- Reimplement TypeScript's APIs in a native speed language
+- Speed Up TypeScript's APIs to native speed
+
+#### Option: Slowing Down to JavaScript Speed
+
+The performance hit option would likely slow the linters down to the point where they have no noticeable performance advantage compared to ESLint.
+👎
+
+That being said, if any native speed linter wants to do this, we in typescript-eslint would love to help.
+The [`@typescript-eslint/parser`](https://typescript-eslint.io/packages/parser) and [`@typescript-eslint/typescript-estree`](https://typescript-eslint.io/packages/typescript-estree) Node.js APIs are open source and as well documented as we've thought to write.
+We'd be happy to work with anybody who wants to use them, including spinning out standalone packages if that'd be useful.
+
+#### Option: Reimplementing TypeScript at Native Speed
+
+Reimplementing TypeScript at native speed is a tantalizing prospect for TypeScript users in general, not just linters.
+I know of three attempts: two actively under development ([Ezno](https://github.com/kaleidawave/ezno) and [`stc`](https://stc.dudy.dev)) and one older one ([TypeRunner](https://github.com/marcj/TypeRunner)).
```

Comment:
Good note from Victorien to clarify: Ezno isn't a drop-in replacement. It has its own type system that attempts to be more powerful, taking advantage of native code speeds.

_(yes, dependent typing!)_